cpp/src/arrow/buffer_builder.h

```diff
@@ -292,6 +292,11 @@ class TypedBufferBuilder<bool> {
     return Status::OK();
   }
 
+  void AppendEmpty(int64_t num_elements) {
```

Comment:
We must not leave elements undefined in an array, because it can reveal previous data. "Empty" values here should still be non-null values of arbitrary value (but still well-defined).

Reply:
Hmm，here is not "AppendEmpty" API, just move null_bit_map pointer to avoid set child arrays null when the top-level struct slot is null. I renamed to "Forward" here.